### PR TITLE
Query Store Service

### PR DESCRIPTION
**Pull request by Benjin**
Adds the Query Store service, which generates queries needed to fetch the data for all query store reports

### CODE REVIEW

**Comment by Benjin** [comment]
Just a draft for now to take a look at how things are working as I finish adding the tests.

At @cheenamalhotra's recommendation, I may end up swapping to instead construct and execute a SqlCommand, and return an ID that ADS can then use to retrieve the result set, as this would remove the need to roll my own .NET -> TSQL datatype conversion code.

File: src/Microsoft.SqlTools.ServiceLayer/QueryStore/QueryStoreService.cs [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+        /// <summary>
+        /// Converts an object (that would otherwise be set as a SqlParameter value) to an entirely TSQL representation.
+        /// Only handles the same subset of object types that Query Store query generators use:
+        /// int, long, string, and DateTimeOffset
+        /// </summary>
+        /// <param name="paramValue"></param>
+        /// <returns>data type and value portions of a parameter declaration, in the form "INT = 999"</returns>
+        internal static string GetTSqlRepresentation(object paramValue)
+        {
+            types.Add(paramValue.GetType());
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/QueryStore/QueryStoreService.cs:525]: do the types added to this HashSet get used somewhere later? I couldn't find where when I did a quick search

File: src/Microsoft.SqlTools.ServiceLayer/QueryStore/Contracts/QueryStoreReportParams.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    public abstract class QueryConfigurationParams<T> : TypedQueryStoreReportParams<T> where T : QueryConfigurationBase, new()
+    {
+        public Metric SelectedMetric;
+        public Statistic SelectedStatistic;
+        public int TopQueriesReturned;
+        public bool ReturnAllQueries;
+        public int MinNumberOfQueryPlans;
+
+        /// <summary>
+        /// Column name by which to order, if any.  Not all query generators involve ordering.
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/QueryStore/Contracts/QueryStoreReportParams.cs:32]: is this comment supposed to go with something else?

Comment (Benjin) [reply src/Microsoft.SqlTools.ServiceLayer/QueryStore/Contracts/QueryStoreReportParams.cs:32]: Yup, something I moved to implementing classes

File: src/Microsoft.SqlTools.ServiceLayer/QueryStore/QueryStoreService.cs [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+        /// Instance of the connection service, used to get the connection info for a given owner URI
+        /// </summary>
+        private ConnectionService ConnectionService { get; }
+
+        private QueryStoreService()
+        {
+            ConnectionService = ConnectionService.Instance;
+        }
+
+        internal QueryStoreService(ConnectionService connService)
```
[the remark this answers is not in the document]
Comment (Benjin) [reply src/Microsoft.SqlTools.ServiceLayer/QueryStore/QueryStoreService.cs:52]: It was originally for testing, but I changed how I was doing that (mocking out the call that needed to run a query instead).

File: src/Microsoft.SqlTools.ServiceLayer/QueryStore/QueryStoreService.cs (hunk 1 above)
Comment (Benjin) [reply src/Microsoft.SqlTools.ServiceLayer/QueryStore/QueryStoreService.cs:541]: Nope; leftover from making sure I'd collected all the different data types that needed to be handled with this code.

File: src/Microsoft.SqlTools.ServiceLayer/QueryStore/Contracts/GetForcedPlanQueriesReport.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+using Microsoft.SqlServer.Management.QueryStoreModel.Common;
+using Microsoft.SqlServer.Management.QueryStoreModel.ForcedPlanQueries;
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+#nullable disable
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryStore.Contracts
+{
+    public class GetForcedPlanQueriesReportParams : QueryConfigurationParams<ForcedPlanQueriesConfiguration>, IOrderableQueryParams
```
Comment (Benjin) [inline src/Microsoft.SqlTools.ServiceLayer/QueryStore/Contracts/GetForcedPlanQueriesReport.cs:14]: TODO: docstring comments for all properties and requests, `get;set;`ers for properties

**Comment by kisantia** [comment]
looks like the other tests also need the line ending change made for the ForcedPlanQueries and TrackedQueries tests, now that they're getting past the timezone failure

**Review by kisantia — APPROVED** [review]

**Pull request merged**